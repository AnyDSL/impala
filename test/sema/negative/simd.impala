fn wrong_type1() -> float {
    simd[1.0f, 2.0f, 3.0f, 4.0f]
}

fn wrong_type2(id: int) -> simd[float * 4] {
    simd[1.0f, 2.0f, 3.0f, 4.0f](id)
}

fn wrong_type3() -> simd[float * 4] {
    simd[1.0f, 2.0f, 3.0f, 4.0f] + simd[1, 2, 3, 4]
}

fn wrong_type4() -> bool {
    simd[true, true, true, true] & true
}

fn wrong_type5() -> simd[bool * 4] {
    simd[true, true, true, true] ^ simd[true, false, true, false, true]
}

fn wrong_type6() -> simd[bool * 4] {
    [true, false, true, false]
}
