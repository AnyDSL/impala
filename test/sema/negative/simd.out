simd.impala:2 col 5 - 32: error: mismatched types: expected 'f32' but found 'simd[f32 * 4]' as return type
simd.impala:6 col 5 - 36: error: mismatched types: expected 'simd[f32 * 4]' but found 'f32' as return type
simd.impala:10 col 36 - 51: error: mismatched types: expected 'simd[f32 * 4]' but found 'simd[i32 * 4]' as return type
simd.impala:14 col 5 - 32: error: mismatched types: expected 'bool' but found 'simd[bool * 4]'
simd.impala:18 col 36 - 71: error: mismatched types: expected 'simd[bool * 4]' but found 'simd[bool * 5]' as return type
simd.impala:22 col 5 - 30: error: mismatched types: expected 'simd[bool * 4]' but found '[bool * 4]' as return type
